File: object/java/src/org/pebblefields/pebbleobject/Callback.java
package org.pebblefields.pebbleobject;

//blocking callback can be used client or server side

public class Callback {

	public Callback(){};

	public void call(Pebble data){};
	public void call(Pebble data, Object idObj){};
	public void call() {
		this.call(null);
	}
}

File: object/csharp/src/Callback.cs
using PebbleFields.PebbleObject;

namespace PebbleFields.PebbleObject
{
    //blocking callback can be used client or server side
    public delegate void Callback(Pebble data);
    public delegate void CallbackWithId(Pebble data, object idObj);
    public delegate void CallbackEmpty();
}
